### PR TITLE
Use ``BitOperations`` in couple more places

### PR DESCRIPTION
**Pull request by PaulusParssinen**
_Should_ be files that are only included by `NetCoreAppToolCurrent` projects.
- **UPDATE1**: Made ILVerification library consumed by ILVerify tool target `NetCoreAppToolCurrent` (same as target as ILVerify)
- **UPDATE2**: Never mind, I didn't realize it was shipped as NuGet package.. Reverting..
<details>
<summary>answered question</summary>
I want to use this to ask, whether any contributions to ``src/coreclr/tools/Common/Internal/NativeFormat/`` is welcomed as the files around here are largely untouched from their initial import and could utilize some newer APIs by my quick skim through. Should these be left alone?
</details>

### CODE REVIEW

File: src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs [hunk 1]
```diff
@@ -2013,16 +2014,10 @@ public void Append(uint hashcode, Vertex element)
...
         private static int HighestBit(uint x)
         {
-            int ret = 0;
-            while (x != 0)
-            {
-                x >>= 1;
-                ret++;
-            }
-            return ret;
+            return 1 << (sizeof(uint) * 8) - BitOperations.LeadingZeroCount(x);
```
Comment (am11) [inline src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs:2020]: nit: to clarify operator precedence:
```suggestion
            return 1 << ((sizeof(uint) * 8) - BitOperations.LeadingZeroCount(x));
```

File: src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs [hunk 2]
```diff
@@ -2051,7 +2046,7 @@ private void ComputeLayout()
             uint bucketsEstimate = (uint)(_Entries.Count / _nFillFactor);
 
             // Round number of buckets up to the power of two
-            _nBuckets = (uint)(1 << HighestBit(bucketsEstimate));
+            _nBuckets = (uint)HighestBit(bucketsEstimate);
```
Comment (huoyaoyuan) [inline src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs:2049]: Should be better to use `RoundUpToPowerOf2`. The old implementation would round up 2^n to next bigger value.

**Comment by MichalStrehovsky** [comment]
> I want to use this to ask, whether any contributions to ``src/coreclr/tools/Common/Internal/NativeFormat/`` is welcomed as the files around here are largely untouched from their initial import and could utilize some newer APIs by my quick skim through. Should these be left alone?

They were not touched because there were no problems with them and they are not perf hotspots. They can be changed. Note that the files with Gen suffix are generated with the https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/Common/Internal/Metadata/NativeFormat/UpdateNativeFormatSources.cmd script so updating them involves updating the generator and re-running it to generate the sources.

File: src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs (hunk 2 above)
Comment (PaulusParssinen) [reply src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs:2049]: I noticed this comment yesterday too, will use that instead.. thanks!

**Comment by PaulusParssinen** [comment]
> They were not touched because there were no problems with them and they are not perf hotspots. They can be changed. Note that the files with Gen suffix are generated with the https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/Common/Internal/Metadata/NativeFormat/UpdateNativeFormatSources.cmd script so updating them involves updating the generator and re-running it to generate the sources.

Okay, thanks for the advice! Unrelated to this PR but should those genered files be perhaps updated to follow the ``.g.cs`` pattern?

File: src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs (hunk 2 above)
Comment (PaulusParssinen) [reply src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs:2049]: Actually, I think the commend here is in accurate that the existing code did not round __up__ to the power of two but only set the highest bit.

e.g.
```csharp
int bucketEstimate = 0b0000_1010;
int highestBitSet = 1 << HighestBit(bucketEstimate); // returns: 0b000_1000
int nextPow2 = BitOperations.RoundUptoPowerOf2(bucketEstimate); // return 0b0001_0000
```

**Comment by MichalStrehovsky** [comment]
> Okay, thanks for the advice! Unrelated to this PR but should those genered files be perhaps updated to follow the `.g.cs` pattern? Or is the contribution bar in these parts: if it works, don't touch? 😄

We're not very consistent with doing this (this is not the only "offline generator") but it might be an improvement to rename them to .g.cs.

File: src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs (hunk 2 above)
Comment (tannergooding) [reply src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs:2049]: Yeah, at a glance it rather looks like the logic for `HighestBit` is incorrect. It claims to be computing the following, where one would normally interpret "rounded up" as  meaning the actual algorithm is `1 + ceil(log2(x)) for non-zero x`
> Returns 1 + log2(x) rounded up, 0 iff x == 0

What's actually happening, however, is that there is no rounding up and it's simply doing `1 + log2(x) for non-zero x`:
```
Console.WriteLine($"{HighestBit(0)} {BitOperations.Log2(0)}");  // 0 0 -- correct
Console.WriteLine($"{HighestBit(1)} {BitOperations.Log2(1)}");  // 1 0 -- correct
Console.WriteLine($"{HighestBit(2)} {BitOperations.Log2(2)}");  // 2 1 -- correct
Console.WriteLine($"{HighestBit(3)} {BitOperations.Log2(3)}");  // 2 1 -- should be 3, as log2(3) is 1.58496...
Console.WriteLine($"{HighestBit(4)} {BitOperations.Log2(4)}");  // 3 2 -- correct
Console.WriteLine($"{HighestBit(5)} {BitOperations.Log2(5)}");  // 3 2 -- should be 4, as log2(5) is 2.321928...
...
```

**Comment by PaulusParssinen** [comment]
I'm unable to build the runtime locally for unknown reasons (#101720) so I'll close this for time being..

**Comment by PaulusParssinen** [comment]
Huh, only asserts in arm32 smoke tests..?

**Comment by MichalStrehovsky** [comment]
> `Expression: 'codeManager->IsUnwindable(pvAddress) || runtime->IsConservativeStackReportingEnabled()'`

This is unlikely to be caused by this PR. We just fixed a bug with similar assert on ARM64. @VSadov do you know if this could also have a manifestation on ARM32 (looks like this CI triggered after your ARM64 fix was merged)?

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/OpenMethodResolver.cs [hunk 3]
```diff
@@ -179,26 +180,20 @@ public static IntPtr ResolveMethod(IntPtr resolverPtr, RuntimeTypeHandle thisTyp
...
-            hash2 = (hash2 + _rotl(hash2, 5)) ^ hashCode4;
+            hash1 = (hash1 + (int)BitOperations.RotateLeft((uint)hash1, 5)) ^ hashCode1;
+            hash2 = (hash2 + (int)BitOperations.RotateLeft((uint)hash2, 5)) ^ hashCode2;
+            hash1 = (hash1 + (int)BitOperations.RotateLeft((uint)hash1, 5)) ^ hashCode3;
+            hash2 = (hash2 + (int)BitOperations.RotateLeft((uint)hash2, 5)) ^ hashCode4;
 
-            hash1 += _rotl(hash1, 8);
-            hash2 += _rotl(hash2, 8);
+            hash1 += (int)BitOperations.RotateLeft((uint)hash1, 8);
+            hash2 += (int)BitOperations.RotateLeft((uint)hash2, 8);
```
Comment (tannergooding) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/OpenMethodResolver.cs:196]: These can just be `int.RotateLeft(hash1, 5)` (and so on)

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeFieldHandle.cs [hunk 4]
```diff
@@ -64,7 +59,7 @@ public override int GetHashCode()
             RuntimeAugments.TypeLoaderCallbacks.GetRuntimeFieldHandleComponents(this, out declaringType, out fieldName);
 
             int hashcode = declaringType.GetHashCode();
-            return (hashcode + _rotl(hashcode, 13)) ^ fieldName.GetHashCode();
+            return (hashcode + (int)BitOperations.RotateLeft((uint)hashcode, 13)) ^ fieldName.GetHashCode();
```
Comment (tannergooding) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeFieldHandle.cs:62]: Same here:
```suggestion
            return (hashcode + int.RotateLeft((hashcode, 13)) ^ fieldName.GetHashCode();
```

File: src/coreclr/tools/Common/Internal/Runtime/EETypeBuilderHelpers.cs [hunk 5]
```diff
@@ -193,15 +194,7 @@ internal static uint ComputeValueTypeFieldPaddingFieldValue(uint padding, uint a
...
-            uint alignmentLog2 = 0;
-            Debug.Assert(alignment != 0);
-
-            while ((alignment & 1) == 0)
-            {
-                alignmentLog2++;
-                alignment >>= 1;
-            }
-            Debug.Assert(alignment == 1);
+            uint alignmentLog2 = (uint)BitOperations.TrailingZeroCount(alignment);
```
Comment (tannergooding) [inline src/coreclr/tools/Common/Internal/Runtime/EETypeBuilderHelpers.cs:197]:
```suggestion
            uint alignmentLog2 = uint.TrailingZeroCount(alignment);
```

File: src/coreclr/tools/Common/TypeSystem/Common/TypeHashingAlgorithms.cs [hunk 6]
```diff
@@ -36,35 +37,29 @@ public void Append(string src)
                 int startIndex = 0;
                 if ((_numCharactersHashed & 1) == 1)
                 {
-                    _hash2 = (_hash2 + _rotl(_hash2, 5)) ^ src[0];
+                    _hash2 = (_hash2 + (int)BitOperations.RotateLeft((uint)_hash2, 5)) ^ src[0];
```
Comment (tannergooding) [inline src/coreclr/tools/Common/TypeSystem/Common/TypeHashingAlgorithms.cs:40]: Basically any of the `BitOperations` APIs where you're having to insert casts can be avoided by just calling the same API directly on the type. Not going to comment on all of them

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/OpenMethodResolver.cs (hunk 3 above)
Comment (PaulusParssinen) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/OpenMethodResolver.cs:196]: Ah, that's much more cleaner. Thanks!

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**